generator/template.go

```diff
@@ -293,7 +293,11 @@ func (td *TemplateData) GenSubSchemas() string {
 				buf.WriteString("*kallax.JSONSchemaArray\n")
 			}
 		} else {
-			buf.WriteString("*kallax.BaseSchemaField\n")
+			if !isRootField(field) && field.IsJSON {
```

Comment:
Comparing your fix with solution for slices...
Does it make sense to apply this fix only for pointers to structs only?

Reply:
I need this function result to match `genSubschemaInit` and everything called before `genSubschemaInit`.
Added a test with struct pointer to ensure it works correctly. https://github.com/veertech/go-kallax/pull/1/commits/690d8601f189a56ca959128ed26bc72224885064